chat-android/src/main/java/com/ably/chat/Presence.kt

```diff
@@ -121,46 +133,80 @@ data class PresenceEvent(
     /**
      * The timestamp of the presence event.
      */
-    val timestamp: Int,
+    val timestamp: Long,
 
     /**
      * The data associated with the presence event.
      */
-    val data: PresenceData,
+    val data: PresenceData?,
 )
 
 internal class DefaultPresence(
-    private val messages: Messages,
+    private val clientId: String,
+    override val channel: Channel,
+    private val presence: PubSubPresence,
 ) : Presence {
 
-    override val channel: Channel
-        get() = messages.channel
-
-    override suspend fun get(params: List<Params>): List<PresenceMember> {
-        TODO("Not yet implemented")
+    override suspend fun get(params: List<Param>): List<PresenceMember> {
+        val usersOnPresence = presence.getCoroutine(params)
+        return usersOnPresence.map { user ->
+            PresenceMember(
+                clientId = user.clientId,
+                action = user.action,
+                data = (user.data as? JsonObject)?.get("userCustomData"),
+                updatedAt = user.timestamp,
+            )
+        }
     }
 
-    override suspend fun isUserPresent(clientId: String): Boolean {
-        TODO("Not yet implemented")
+    override suspend fun get(waitForSync: Boolean, clientId: String?, connectionId: String?): List<PresenceMember> {
+        val params = buildList {
+            if (waitForSync) add(Param(GET_WAITFORSYNC, true))
```

Comment:
Having an integration test would have been useful : |

Reply:
I can make this change of course, but it will make no difference, if value type is not  `String`, `Param` constructor implicitly invoke `toString()`. I just think it's looks more appropriate that we send boolean here. As you mentioned it will be deserialized later

Reply:
Thanks for the info 👍